### DISCUSSION TITLE
Rabbitmq version upgrade data loss in kubernetes cluster

### DISCUSSION THREAD

**Discussion by paywagh11** [rabbitmq/rabbitmq-server#9152] [Questions]
When we are upgrading the RMQ cluster(pods) version to newer version, in the middle of the publisher is publishing the messages , some of the messages are lost even though we have enabled durable queues and persistent . Means if publisher is sending 10k messages only 8k messages are stored in the queue during the upgrade activity.

What can we do to get all the data published by publisher at consumer end?

**Comment by mkuratczyk** [comment]
Use publisher confirms (and consider how many in-flight messages your are ok with), and the mandatory flag.

https://www.rabbitmq.com/publishers.html#data-safety
https://www.rabbitmq.com/publishers.html#unroutable

If you can reproduce this problem with https://perftest.rabbitmq.com/, please provide the flags (then we'll know how exactly the publisher is set up)
